docs/import/add_disaster.js

```diff
@@ -0,0 +1,253 @@
+    // pickers from previous disasters don't hang around.
+    assetPickerDiv.empty();
+    return assetPickersDone;
+  }
+}
+
+/**
+ * Writes the given details to a new disaster entry in firestore. Fails if
+ * there is an existing disaster with the same details.
+ * @param {string} disasterId of the form <year>-<name>
+ * @param {Array<string>} states array of state (abbreviations)
+ * @return {Promise<boolean>} returns true after successful write to firestore.
+ */
+function writeNewDisaster(disasterId, states) {
+  if (disasters.has(disasterId)) {
+    setStatus('Error: disaster with that name and year already exists.');
+    return Promise.resolve(false);
+  }
+
+  disasters.set(disasterId, states);
+  toggleState(disasterId);
+
+  clearStatus();
+  const disasterOptions = $('#disaster > option');
+  // comment needed to quiet eslint on no-invalid-this rules
+  disasterOptions.each(/* @this HTMLElement */ function() {
+    if ($(this).val() > disasterId) {
+      $(createOptionFrom(disasterId)).insertBefore($(this));
+      return false;
+    }
+  });
+  $('#disaster').val(disasterId);
+  return getFirestoreRoot()
+      .collection('disaster-metadata')
+      .doc(disasterId)
+      .set({states: states})
+      .then(() => true);
+}
+
+/**
+ * Onclick function for the new disaster form. Writes new disaster to firestore,
+ * local disasters map and disaster picker. Doesn't allow name, year or states
+ * to be empty fields.
+ * @return {Promise<boolean>} resolves true if new disaster was successfully
+ *     written.
+ */
+function addDisaster() {
+  const year = $('#year').val();
+  const name = $('#name').val();
+  const states = $('#states').val();
+
+  if (!year || !name || !states) {
+    setStatus('Error: Disaster name, year, and states are required.');
+    return Promise.resolve(false);
+  }
+  if (isNaN(year)) {
+    setStatus('Error: Year must be a number.');
+    return Promise.resolve(false);
+  }
+  const disasterId = year + '-' + name;
+  return writeNewDisaster(disasterId, states);
+}
+
+// Needed for testing :/
+const emptyCallback = () => {};
+
+// TODO: add functionality for adding assets to disaster records from these
+// pickers.
+/**
+ * Requests all assets in ee directories corresponding to given states.
+ * @param {Array<string>} states e.g. ['WA']
+ * @return {Promise<void>} completes when we've finished filling all state
+ * pickers.
+ */
+function getAssetsFromEe(states) {
+  return ee.data.listAssets(eeLegacyPathPrefix + 'states', {}, emptyCallback)
+      .then((result) => {
+        const folders = new Set();
+        for (const folder of result.assets) {
+          folders.add(folder.id.substring((gdEePathPrefix + 'states/').length));
+        }
+        const promises = [];
+        for (const state of states) {
+          const dir = eeLegacyPathPrefix + 'states/' + state;
+          if (!folders.has(state)) {
+            ee.data.createFolder(dir, false, emptyCallback);
```

Comment:
can we tell when this create call finishes?

Reply:
We can give it a alert a butter bar or something in the callback. But I'd prefer for the user not to have to wait for folder creation to see the empty select. I've also added the set acls call so that's where the status update can go too.